ibtk/src/math/HierarchyMathOps.cpp

```diff
@@ -2003,30 +2003,64 @@ HierarchyMathOps::interp(const int dst_idx,
 
 void
 HierarchyMathOps::interp(const int dst_idx,
-                         const Pointer<NodeVariable<NDIM, double> > /*dst_var*/,
-                         const bool dst_ghost_interp,
+                         const Pointer<NodeVariable<NDIM, double> > dst_var,
+                         const bool dst_cf_bdry_synch,
                          const int src_idx,
                          const Pointer<CellVariable<NDIM, double> > /*src_var*/,
                          const Pointer<HierarchyGhostCellInterpolation> src_ghost_fill,
                          const double src_ghost_fill_time)
 {
     if (src_ghost_fill) src_ghost_fill->fillData(src_ghost_fill_time);
 
-    for (int ln = d_coarsest_ln; ln <= d_finest_ln; ++ln)
+    Pointer<NodeDataFactory<NDIM, double> > data_factory = dst_var->getPatchDataFactory();
+    const int depth = data_factory->getDefaultDepth();
+    if (dst_cf_bdry_synch)
+    {
+        TBOX_ASSERT(depth == 1 || depth == NDIM);
+    }
+    const int on_idx = depth == 1 ? d_on_s_idx : d_on_v_idx;
+
+    for (int ln = d_finest_ln; ln >= d_coarsest_ln; --ln)
     {
         Pointer<PatchLevel<NDIM> > level = d_hierarchy->getPatchLevel(ln);
 
-        // Interpolate.
+        // Allocate temporary data to synchronize the coarse-fine interface.
+        if (ln > d_coarsest_ln && dst_cf_bdry_synch)
+        {
+            level->allocatePatchData(on_idx);
+        }
+
+        // Interpolate and extract data on the coarse-fine interface.
         for (PatchLevel<NDIM>::Iterator p(level); p; p++)
         {
             Pointer<Patch<NDIM> > patch = level->getPatch(p());
 
             Pointer<NodeData<NDIM, double> > dst_data = patch->getPatchData(dst_idx);
             Pointer<CellData<NDIM, double> > src_data = patch->getPatchData(src_idx);
 
-            d_patch_math_ops.interp(dst_data, src_data, patch, dst_ghost_interp);
+            // We don't want to fill ghosts here - that will be done in another way
+            d_patch_math_ops.interp(dst_data, src_data, patch, false);
```

Comment:
I don't know what it means to fill ghost cells in a patch operator. This probably needs to be changed in `PatchMathOps`.

Reply:
It looks like this flag extends the averaging region into the ghost cell region. This is confusing because it is not how most other patch or hierarchy math operators are set up. It looks like the variable-coefficient solver relies on this.

I can see the argument for evaluating on a larger box than the patch box --- it requires a larger volume of communication but only one ghost filling step. Perhaps there could be a configuration flag in the constructors for both `HierarchyMathOps` and `PatchMathOps` that control whether operations are evaluated on the patch box, or in the largest box for which ghost cells are available.

Reply:
#1524 provides a possible way to go here.